### DISCUSSION TITLE
template in module

### DISCUSSION THREAD

**Discussion by ahsf** [cue-lang/cue#1936] [Q&A] [answered]
How can I move a template into a module / package?

all in one file works.
```
job: [Name=_]: {
    name:     Name
    replicas: uint | *1
    command:  string
}

job: list: command: "ls"

job: nginx: {
    command:  "nginx"
    replicas: 2
}
```

I tried this:

job.cue
```
package job

[Name=_]: {
    name:     Name
    replicas: uint | *1
    command:  string
}
```

but the template is not used when I import it.

```
import "domain.com/pkgjob"

job: list: command: "ls"

job: nginx: {
    command:  "nginx"
    replicas: 2
}
```

**Comment by eadlam** [comment, marked as answer]
**job.cue**
```cue
package job

[Name=_]: {
    name:     Name
    replicas: uint | *1
    command:  string
    test: "hello"
}
```

**main.cue**
```cue
package main

import Job "domain.com/job"

job: Job

job: list: command: "ls"

job: nginx: {
    command:  "nginx"
    replicas: 2
}
```

**Comment by ahsf** [comment]
this works, thank you.

I thought `import "domain.com/job"` would already pull a job object into the using file. So just lke copy paste into the file, but that seems not to be true.
There is not much info about whats going on with modules. When I do

```
import Job "domain.com/job"
```

What is `Job`?
Is it something I can construct without a module or is it something special?

**Reply by eadlam** [reply]
When you import package **job** and then also have a top-level item named **job**, you should get an error like this:

```
job redeclared as imported package name
```

To avoid that conflict, you can rename the imported package **job** to something else. That's what this does:

```cue
import Job "domain.com/job"
```

This imports package **job** under the alias **Job**.
